test/e2e/es/version_upgrade_test.go

```diff
@@ -207,6 +207,7 @@ func TestVersionUpgradeTwoNodesToLatest8x(t *testing.T) {
 
 	initial := elasticsearch.NewBuilder("test-version-up-2-to-8x").
 		WithVersion(srcVersion).
+		WithChangeBudget(2, 2). // 8x non-HA upgrades cannot honour a change budget with maxUnavailable 1
```

Comment:
I would explain in the comment that this is to "circumvent" the [`changeBudgetWatcher`](https://github.com/elastic/cloud-on-k8s/blob/780831adab3c88e57666d7a1659a589a4a2b5d70/test/e2e/test/elasticsearch/steps_mutation.go#L83). Having a change budget with `maxUnavailable` set to 1 does not prevent the operator to restart all the Pods in this particular case (which should maybe stated in the [documentation](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-update-strategy.html#k8s-update-strategy) ?)

Reply:
I am going to do the documentation update in a separate PR to get this merged today.